File: whitefilterstr.php
<?php

/**
 * Filter a string and allow only characters in the white list string
 *
 * @param string $inpStr
 * @param string $whiteListStr Allowed characters
 * @return string If fault then empty
 */
 function whiteFilterStr($inpStr, $whiteListStr) {
  $strClean = '';
  
  if (!is_string($inpStr) or !is_string($whiteListStr)) {
     return $strClean;
    }
  $inpStrLen = strlen($inpStr);
  # Filter the input string with the whitelist
  $i = 0;
  while ($i < $inpStrLen) {
    if (strpos($whiteListStr, $inpStr{$i}, 0) !== false) {
      $strClean = $strClean.$inpStr{$i};
    }
  $i++;
  }
  return $strClean;
}
 ?>

File: whitefilterstr.py
def whiteListCharFilter(inStr, whiteListStr):
    """ Sanitize a string with a list of allowed (white) characters

    Input: inStr {string} String to be sanitized.
    Input: whiteListStr {string} String with allowed characters.
    Output: outStr {string} Sanitized string
    """
    outStr = ""
    if (isinstance(inStr, str) or isinstance(whiteListStr, str)) == False:
        return outStr
    for characterStr in inStr:
        if characterStr in whiteListStr:
            outStr = outStr + characterStr
    return outStr
    
